src/shared/settingsConfiguration.ts

```diff
@@ -56,4 +56,50 @@ export class DefaultSettingsConfiguration implements SettingsConfiguration {
             return false
         }
     }
+    /**
+     * Sets a prompt message as suppressed.
+     * @param promptName Name of prompt to append to list
+     */
+    public disable(promptName: string): void {
+        const setting = this.getSuppressPromptSetting(promptName)
+        if (setting === undefined || setting[promptName]) {
+            return
+        }
+        setting[promptName] = true
+        this.writeSetting('suppressPrompts', setting, vscode.ConfigurationTarget.Global)
+    }
+    /**
+     * Verifies if a prompt should be displayed again.
+     * @param promptName Name of the prompt
+     * @returns False when prompt has been suppressed
+     */
+    public shouldDisplayPrompt(promptName: string): boolean {
+        const promptSetting = this.getSuppressPromptSetting(promptName)
+        if (promptSetting !== undefined && promptSetting[promptName]) {
+            return false
+        }
+        return true
+    }
+
+    /**
+     * Gets the 'suppressPrompts' settings object. This will reset the setting to default if it does not
+     * recieve an object or the prompts value is not a boolean.
+     * @param promptName
+     * @returns The settings object
+     */
+    public getSuppressPromptSetting(promptName: string): { [prompt: string]: boolean } | undefined {
+        try {
+            const setting = this.readSetting<{ [prompt: string]: boolean }>('suppressPrompts')
+            if (typeof setting !== 'object' || typeof setting[promptName] !== 'boolean') {
```

Comment:
condition is too loose. if `setting` is undefined we don't need to enter this if-block and rewrite the settings. That would result in forcibly adding the setting even for users who don't disable the prompts. It's also a small perforfmance cost.
```suggestion
            if (setting !== undefined && (typeof setting !== 'object' || typeof setting[promptName] !== 'boolean')) {
```

Reply:
This is a good point and the rest of the flow already allows for undefined values, so this change makes sense.

Reply:
> UPDATE: I don't think `setting` can ever be undefined.

Then we should change the return type of `readSetting()` (separate PR).